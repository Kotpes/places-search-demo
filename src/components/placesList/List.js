//@flow

import * as React from 'react'
import {Link} from '@reach/router'
import type {Venue} from './PlacesList'

type Locations = {
  locations: Array<Venue>
}

const List = ({locations}: Locations) => {
  return locations.map(location => {    
    const {id, name} = location
    
    return (
      <li key={id}>
        <Link to={`places/${id}`}>{name}</Link>
      </li>   
    )
  })
}


export default List