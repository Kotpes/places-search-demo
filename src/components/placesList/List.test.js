import React from 'react'
import TestRenderer from 'react-test-renderer'
import {Link} from '@reach/router'
import List from './List'

it(`Matches the snapshot`, () => {
  const locations = [
    {id: 1, name: 'Someplace', location: {distance: 100, address: 'Some address', lat: 70.21, lng: 29.01}}
  ]
  const component = TestRenderer.create(<List locations={locations}/>)  
  expect(component.toJSON()).toMatchSnapshot()
})

it(`Contains link to the place`, () => {
  const locations = [
    {id: 1, name: 'Someplace', location: {distance: 100, address: 'Some address', lat: 70.21, lng: 29.01}}
  ]
  const component = TestRenderer.create(<List locations={locations}/>) 
  //Link is rendered 
  expect(component.root.findByType(Link)).toBeDefined()
  //Link has correct to prop value
  expect(component.root.findByType(Link).props.to).toBe('places/1') 
})