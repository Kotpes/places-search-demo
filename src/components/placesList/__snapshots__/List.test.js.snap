// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot 1`] = `
<li>
  <a
    href="/places/1"
    onClick={[Function]}
  >
    Someplace
  </a>
</li>
`;
